src/context/AgentContext.tsx

```diff
@@ -0,0 +1,295 @@
+import { showNotification } from "@mantine/notifications";
+import { Ad4mClient, ExceptionType, Link } from "@perspect3vism/ad4m";
+import { ExceptionInfo } from "@perspect3vism/ad4m/lib/src/runtime/RuntimeResolver";
+import { createContext, useEffect, useState } from "react";
+import { AD4M_ENDPOINT } from "../config";
+import { SOURCE_PROFILE, PREDICATE_FIRSTNAME, PREDICATE_LASTNAME } from "../constants/triples";
+import { buildAd4mClient } from "../util";
+
+type State = {
+  url: string;
+  isInitialized: Boolean;
+  isUnlocked: Boolean;
+  loading: boolean;
+  client: Ad4mClient | null;
+  did: string,
+  candidate: string;
```

Comment:
`candidate` should not be put in a global state, it's only used when adding trusted agent.